### PR TITLE
chore: add verbose logs for latency and operation start timestamps

### PR DESCRIPTION
**Pull request by feiskyer**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

chore: add verbose logs for latency and operation start timestamps

Sample logs:

```
I0314 12:57:28.635713       1 azure_metrics.go:113] "Observed Request Latency" latency_seconds=27.475052838 request="services_ensure_loadbalancer_deleted" resource_group="ccm" subscription_id="<subscription>" source="default/http" result_code="succeeded"
I0314 12:57:59.014313       1 azure_metrics.go:113] "Observed Request Latency" latency_seconds=0.08678909 request="services_vmss_ensure_hosts_in_pool" resource_group="ccm" subscription_id="<subscription>" source="default/http" result_code="succeeded"
I0314 12:58:03.099604       1 azure_metrics.go:113] "Observed Request Latency" latency_seconds=9.610200225 request="services_ensure_loadbalancer" resource_group="ccm" subscription_id="<subscription>" source="default/http" result_code="succeeded"
```

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
chore: add verbose logs for latency and operation start timestamps
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/metrics/azure_metrics.go [hunk 1]
```diff
@@ -81,22 +82,35 @@ func (mc *MetricContext) ThrottledCount() {
...
 	if rerr != nil {
 		errorCode := rerr.ServiceErrorCode()
 		attributes := append(mc.attributes, errorCode)
 		apiMetrics.errors.WithLabelValues(attributes...).Inc()
 	}
+	mc.logLatency(latency, "error_code", rerr.ServiceErrorCode())
 }
 
 // ObserveOperationWithResult observes the request latency and failed requests of an operation.
 func (mc *MetricContext) ObserveOperationWithResult(isOperationSucceeded bool) {
```
Comment (andyzhangx) [inline pkg/metrics/azure_metrics.go:96]: add a new parameter(`operation string`) in `ObserveOperationWithResult` and then in L104, write like `mc.logLatency(latency, operation, resultCode)`

Comment (feiskyer) [reply pkg/metrics/azure_metrics.go:96]: Those metadata would be logged (including the request -> request is the operation):

```go
metricLabels = []string{
		"request",         // API function that is being invoked
		"resource_group",  // Resource group of the resource being monitored
		"subscription_id", // Subscription ID of the resource being monitored
		"source",          // Operation source(optional)
	}

```

**Comment by andyzhangx** [comment]
/hold
hold a moment for the change

File: pkg/metrics/azure_metrics.go [hunk 2]
```diff
@@ -80,23 +81,35 @@ func (mc *MetricContext) ThrottledCount() {
...
 		apiMetrics.errors.WithLabelValues(attributes...).Inc()
 	}
+	mc.logLatency(latency, append(labelAndValues, "error_code", rerr.ServiceErrorCode())...)
 }
 
 // ObserveOperationWithResult observes the request latency and failed requests of an operation.
-func (mc *MetricContext) ObserveOperationWithResult(isOperationSucceeded bool) {
-	operationMetrics.operationLatency.WithLabelValues(mc.attributes...).Observe(
-		time.Since(mc.start).Seconds())
+func (mc *MetricContext) ObserveOperationWithResult(isOperationSucceeded bool, labelAndValues ...interface{}) {
```
Comment (feiskyer) [inline pkg/metrics/azure_metrics.go:96]: @andyzhangx added `labelAndValues` for ObserveOperationWithResult, which allows pass in label and values together (e.g. `ObserveOperationWithResult(result, "object_name", "name", "object_namespace", "namespace")`).

PTAL

**Review by andyzhangx — APPROVED** [review]
/lgtm
/retest

**Pull request merged**